core/src/main/java/hudson/PluginManager.java

```diff
@@ -1923,6 +1924,31 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
         return FormValidation.ok();
     }
 
+    @Restricted(NoExternalUse.class)
+    @RequirePOST public FormValidation doCheckUpdateSiteUrl(StaplerRequest request, @QueryParameter String value) {
```

Comment:
This needs to be protected by a check to ensure the user has admin permissions to prevent server side request forgery.

```suggestion
    @RequirePOST public FormValidation doCheckUpdateSiteUrl(StaplerRequest request, @QueryParameter String value) {
        Jenkins.get().checkPermission(Jenkins.ADMINISTER);
```

Reply:
Yess, this should be protected. Thanks for the suggestion!